backends-velox/src/main/scala/org/apache/gluten/extension/AppendBatchResizeAfterShuffleRead.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.extension
+
+import org.apache.gluten.config.VeloxConfig
+import org.apache.gluten.execution.VeloxResizeBatchesExec
+
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.{ColumnarShuffleExchangeExec, SparkPlan}
+import org.apache.spark.sql.execution.adaptive.{AQEShuffleReadExec, ShuffleQueryStageExec}
+
+case class AppendBatchResizeAfterShuffleRead() extends Rule[SparkPlan] {
```

Comment:
We don't have a rule for the resizing on the input side of shuffle. Can we make the ways of optimizations more consistent? Either both via rules, or both not?

Reply:
At first,  resizing on the output side of shuffle followings the way for shuffle read, that's do it when converting to transformer. But after the `DummpyLeafExec` is introduced, it doesn't work.
So I'll refactor the way to add resizing on the input side of shuffle, to make it enabled by rule

Reply:
Fair enough. Let's make it an individual rule.

Reply:
update